### ISSUE TITLE
Doesn't handle templates inside quotes

### ISSUE THREAD

**Issue by chocolateboy** [medikoo/es6-template-strings#3]
`${"foo"}` works, but `${"${foo}"}` and `${'${foo}'}` don't.

> SyntaxError: Unexpected token ILLEGAL

I would expect the latter two to produce `"${foo}"`.

### Test

```javascript
var assert = require('assert');
var Template = require ('es6-template-strings');

assert.strictEqual(Template('${"${foo}"}'), '${foo}');
assert.strictEqual(Template("${'${foo}'}"), '${foo}');
```

**Comment by medikoo** [comment]
@chocolateboy thanks for reporting. Indeed it's wrong. Reason is that currently this util uses very simple parser, which does not parse internal code as JS code. Fix for that, would be to rely on something as https://github.com/medikoo/esniff to parse code part.

**Comment by medikoo** [comment]
Fixed with v2.0.0
